data-plane/receiver/src/main/java/dev/knative/eventing/kafka/broker/receiver/CloudEventRequestToRecordMapper.java

```diff
@@ -19,34 +19,53 @@
 import static net.logstash.logback.argument.StructuredArguments.keyValue;
 
 import dev.knative.eventing.kafka.broker.core.cloudevents.PartitionKey;
+import dev.knative.eventing.kafka.broker.core.tracing.Tracing;
+import dev.knative.eventing.kafka.broker.core.tracing.TracingSpan;
 import io.cloudevents.CloudEvent;
 import io.cloudevents.core.message.MessageReader;
 import io.cloudevents.http.vertx.VertxMessageFactory;
 import io.vertx.core.Future;
+import io.vertx.core.Vertx;
 import io.vertx.core.http.HttpServerRequest;
 import io.vertx.kafka.client.producer.KafkaProducerRecord;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 public class CloudEventRequestToRecordMapper implements RequestToRecordMapper<String, CloudEvent> {
 
-  private static final Logger logger = LoggerFactory
-    .getLogger(CloudEventRequestToRecordMapper.class);
+  private static final Logger logger = LoggerFactory.getLogger(CloudEventRequestToRecordMapper.class);
+
+  private final Vertx vertx;
+
+  public CloudEventRequestToRecordMapper(final Vertx vertx) {
+    this.vertx = vertx;
+  }
 
   @Override
   public Future<KafkaProducerRecord<String, CloudEvent>> recordFromRequest(
     final HttpServerRequest request,
     final String topic) {
 
     return VertxMessageFactory.createReader(request)
-      // TODO is this conversion really necessary?
-      //      can be used Message?
       .map(MessageReader::toEvent)
       .map(event -> {
         if (event == null) {
           throw new IllegalArgumentException("event cannot be null");
         }
-        logger.debug("received event {}", keyValue("event", event));
+
+        if (logger.isDebugEnabled()) {
+          final var span = TracingSpan.getCurrent(vertx);
+          if (span != null) {
+            logger.debug("received event {} {}",
+              keyValue("event", event),
+              keyValue(Tracing.TRACE_ID_KEY, span.getSpanContext().getTraceIdAsHexString())
+            );
+          } else {
+            logger.debug("received event {}", keyValue("event", event));
+          }
+        }
+
+        TracingSpan.decorateCurrent(vertx, event);
```

Comment:
So this is all we need to set the span in kafka? cool!

Reply:
This is specifically to add business context to the span (cloudevents attributes in this case).

Reply:
Ok and then the span propagation will be handled transparently behind the hood right?

Reply:
Yes, Vertx calls sendRequest for this.

The flow is `sendRequest` -> `decorateCurrent` -> `receiveResponse` for both `WebClient` and `KafkaProducer`